src/context/WalletProvider/components/WalletConnectTopSection.tsx

```diff
@@ -0,0 +1,95 @@
+import { Box, Button, Divider, HStack, Stack } from '@chakra-ui/react'
+import type EthereumProvider from '@walletconnect/ethereum-provider'
+import { useCallback, useState } from 'react'
+
+import { WalletConnectDirectRow } from '../WalletConnectV2/components/WalletConnectDirectRow'
+
+import { Text } from '@/components/Text'
+import { WalletActions } from '@/context/WalletProvider/actions'
+import { KeyManager } from '@/context/WalletProvider/KeyManager'
+import { useLocalWallet } from '@/context/WalletProvider/local-wallet'
+import { WalletConnectV2Config } from '@/context/WalletProvider/WalletConnectV2/config'
+import { WalletNotFoundError } from '@/context/WalletProvider/WalletConnectV2/Error'
+import { useWallet } from '@/hooks/useWallet/useWallet'
+import { isWalletConnectWallet } from '@/lib/utils'
+import { clearWalletConnectLocalStorage } from '@/plugins/walletConnectToDapps/utils/clearAllWalletConnectToDappsSessions'
+
+export const WalletConnectTopSection = () => {
+  const { dispatch, state, getAdapter } = useWallet()
+  const localWallet = useLocalWallet()
+  const [loading, setLoading] = useState(false)
+
+  const pairDevice = useCallback(async () => {
+    clearWalletConnectLocalStorage()
+    setLoading(true)
+    dispatch({ type: WalletActions.SET_WALLET_MODAL, payload: false })
+
+    const adapter = await getAdapter(KeyManager.WalletConnectV2)
+
+    try {
+      if (adapter) {
+        if (!state.wallet || !isWalletConnectWallet(state.wallet)) {
```

Comment:
If already connected to a WC wallet, this won't work proper - not an issue introduced by this PR, but just noting this so I tackle it as a follow-up of this PR.